### DISCUSSION TITLE
Intellij Plugin not Showing Up

### DISCUSSION THREAD

**Discussion by Seggan** [community/community#6619] [Copilot Conversations]
**Description**:
When I follow the steps to get the plugin on Intellij IDEA Community, it does not show up

**Steps to reproduce**:
- Add the plugin repository
- See no Github Copilot

**Expected behavior**:
Github Copilot to show up

![image](https://user-images.githubusercontent.com/49963543/138519699-440e6b6b-4631-4ec8-9109-148df0b033a4.png)

**Comment by hsenag** [comment]
What version of IntelliJ are you using? We require `2021.2` or above.

**Reply by uwueviee** [reply]
I'm also having this issue and I have tried it on 2021.2.3 and 2021.3 EAP

**Reply by ffMathy** [reply]
Same as @uwueviee. For instance, it can't be installed nor found with Rider, latest version.

**Reply by jkenneydaniel** [reply]
I don't understand why this version requirement isn't documented on the JetBrains setup guide :( - it would be helpful to know before spending an hour trying to figure out why adding the repository just results in "No plugins found".

**Reply by pgrennin** [reply]
I had this issue and can confirm that updating Intellij to latest version and it shows up

**Reply by FeelsBadMan1** [reply]
> I had this issue and can confirm that updating Intellij to latest version and it shows up

Thats not the case on other IDEs. Im a Student and got Github Education Free so i got the Toolbox and all IDEs regardless of Pro or full version free however updating Intellij shows the plugin within Intellij but not across other IDEs such as Pycharm and i am a Python Programmer so this would be useful. I also tried to copy the plugin.zip to the pycharm dictonary and renamed the Github_copilot_Intellij.zip to Github_copilot_Pycharm.zip but this sadly not changed anything and its still not showing up.

**Reply by aspearman** [reply]
I found the plugin on WebStorm 2021.2.3 (the latest) just now

**Reply by thangvd2** [reply]
I also cannot find the plugin in ultimate 2021.2.3 
<img width="1292" alt="image" src="https://user-images.githubusercontent.com/7755500/139032058-55f7811f-69aa-4681-8f4b-d9ffd5112208.png">

**Reply by hsenag** [reply]
> I don't understand why this version requirement isn't documented on the JetBrains setup guide :( - it would be helpful to know before spending an hour trying to figure out why adding the repository just results in "No plugins found".

Sorry about that, we should have done so from the beginning. We'll be making various updates later today that should include that information.

**Reply by jansorg** [reply]
@thangvd2 Your version is supported. Try restarting the IDE after adding the channel to force a refresh of the sources.

**Comment by Seggan** [comment]
Ah I have 2021.1. Will try and report the result

**Comment by actioninja** [comment]
Alright after some experimentation and research. I figured out some things:

The URL is of an outdated format and redirects to the correct one. Inputting the url in a browser directs you to `https://plugins.jetbrains.com/plugins/list?channel=super-early-bird`, which contains an API error response saying it needs a build. Intellij itself adds this parameter, and so (with Intellij 2021.2.3) the plugin shows up as expected using that url.

The repository only shows up on Intellij. The build parameter passed by CLion results in the API responding with an empty repository which CLion considers to be an error. However, you can actually get the plugin running and working on any jetbrains IDE by installing it on Intellij and then copying the plugin file to the other IDE's plugin folder.

[Here's some jetbrains docs about where plugins live.](https://intellij-support.jetbrains.com/hc/en-us/articles/206544519-Directories-used-by-the-IDE-to-store-settings-caches-plugins-and-logs) If you use toolbox, look in the local folder instead of the roaming folder.

**Reply by lingsamuel** [reply]
I got error ``IDE contains conflicting module `com.intellij.modules.rider` `` in Rider.

**Reply by A-z-z-y** [reply]
Rider isn't actually officially supported, but It'll still work for a while before that happens. Just reinstall and it'll work again. @lingsamuel

**Reply by michlimlim** [reply]
I got the same error but in Clion: `Plugin 'GitHub Copilot' is not compatible with the IDE because the IDE contains module 'com.intellij.modules.clion' which conflicts with the plugin`

What do you mean reinstall?

**Reply by ffMathy** [reply]
@michlimlim also experiencing that, but on Rider instead.

**Reply by TechnicallyMay** [reply]
I was able to "fix" this by opening "~\AppData\Roaming\JetBrains\Rider2021.2\plugins\github-copilot-intellij\lib\github-copilot-intellij-0.18.0.jar\META-INF\plugin.xml" and removing this line:

`<incompatible-with>com.intellij.modules.clion</incompatible-with>`

**Reply by ffMathy** [reply]
Wow interesting! Makes me wonder why they explicitly made it incompatible with IDEs where it obviously just works.

**Reply by Vahalaru** [reply]
Would this also work for Android Studio?

**Reply by ffMathy** [reply]
@TechnicallyMay this doesn't actually work for me. Did you do anything else?

**Comment by dmwyatt** [comment]
Not to say anyone is not having legit issues, but FWIW, I had no problems installing the plugin [as directed](https://github.com/github/copilot-docs/blob/main/docs/gettingstartedjetbrains.md).

Make sure you're searching in the "Marketplace" section.

**Comment by BorkforceOne** [comment]
Can't seem to get this to work with the latest version of Rider available (2021.2.2). Is only a subset of Intellij products supported here?

**Reply by A-z-z-y** [reply]
I was also unable to find it in Rider, even on the EAP version.

**Reply by ecarandang** [reply]
Same here. Can't make it to work with Rider.

**Reply by FeelsBadMan1** [reply]
Same with Pycharm

**Comment by Seggan** [comment]
Update, I updated my Intellij and it worked as expected 👍

**Reply by ffMathy** [reply]
This does not work for Rider or other IDEs.

**Reply by Longer430** [reply]
Does not work for Pycharm

**Comment by benjamin-ict** [comment]
Looking at the plugin home (https://plugins.jetbrains.com/plugin/17718-github-copilot) it says that the plugin does not support Rider or CLion as of now: 

> Compatible with IntelliJ IDEA (Ultimate, Community, Educational), Android Studio, Code With Me Guest, DataGrip, DataSpell, GoLand, MPS, PhpStorm, PyCharm (Professional, Community, Educational), RubyMine, WebStorm

**Reply by ffMathy** [reply]
This can be worked around though by modifying the Metadata of the plugin (see the other thread in here), but that just makes it even weirder that they explicitly excluded support for it, when it clearly works.

**Reply by FeelsBadMan1** [reply]
That is weird because it also not shows on Pycharm

**Comment by FeelsBadMan1** [comment]
same issue i having on pycharm professional and the pycharm ide is not even incompatible inside the plugin.xml file and its still not shown. i tried to install it on intellji and copy the .zip from the pluginfolder to the folder from pycharm but it was still not shown.

**Reply by jansorg** [reply]
Which version of PyCharm are you using? 2021.2 and later is needed.

**Reply by FeelsBadMan1** [reply]
> Which version of PyCharm are you using? 2021.2 and later is needed.

I use 2021.1 i think but when checking for updates no update is avaible.

**Reply by jansorg** [reply]
You can check your version at `Help > About` (in the main menu on macOS). 2021.2 has been published by JetBrains, so you might have to install manually or with your package manager.

**Reply by FeelsBadMan1** [reply]
I use Windows and Toolbox

**Reply by Longer430** [reply]
I have my Pycharm updated to the latest version.

**Reply by jansorg** [reply]
I'm sorry, but I don't know why it's not showing up. PyCharm 2021.2 or later is supported. If it's not showing up, try to refresh the plugin sources (e.g. by restarting) and that your network connectivity is okay. If nothing helps, download the latest version from https://plugins.jetbrains.com/plugin/17718-github-copilot/versions and drag&drop it onto PyCharm to install it.
